server/src/rbac.rs

```diff
@@ -16,136 +16,139 @@
  *
  */
 
-use std::sync::RwLock;
+use std::sync::{RwLock, RwLockReadGuard, RwLockWriteGuard};
 
 use once_cell::sync::OnceCell;
 
+use crate::option::CONFIG;
+
 use self::{
-    role::{Action, Permission},
-    user::{get_admin_user, verify, User, UserMap, UserPermMap},
+    auth::AuthMap,
+    role::{model::DefaultPrivilege, Action},
+    user::{get_admin_user, User, UserMap},
 };
 
+pub mod auth;
 pub mod role;
 pub mod user;
 
-pub static USER_AUTHENTICATION_MAP: OnceCell<RwLock<UserMap>> = OnceCell::new();
-pub static USER_AUTHORIZATION_MAP: OnceCell<RwLock<UserPermMap>> = OnceCell::new();
+pub static USER_MAP: OnceCell<RwLock<UserMap>> = OnceCell::new();
+pub static AUTH_MAP: OnceCell<RwLock<AuthMap>> = OnceCell::new();
+
```

Comment:
What is the purpose of two different maps here?

Reply:
If any changes made to user ( password change, role update, etc.. ) it is updated in memory in the Usermap `Hashmap<username, User>`. The User contains password hash and role information. While `AuthMap` is for quick authentication and authorization, it is HashMap<(user, pass), Vec<Permission>>. Any changes made to User will reset the AuthMap and upon authentication user perms is reloaded into this map